astroid/interpreter/_import/spec.py

```diff
@@ -238,6 +238,16 @@ def find_module(self, modname, module_parts, processed, submodule_path):
                 type=module_type,
                 submodule_search_locations=list(spec.submodule_search_locations or []),
             )
+        elif modname in sys.builtin_module_names:
+            # PyPy does not necessarily supply a __spec__
+            # https://foss.heptapod.net/pypy/pypy/-/issues/3736#note_184637
+            spec = ModuleSpec(
+                name=modname,
+                location=None,
+                origin="built-in",
```

Comment:
Is there a reason why we add `origin` and `submodule_search_locations` here? We don't do it for the other `builtins` `ModuleSpec`.

Reply:
Thanks, you have a point that I could move this to the `ImportlibFinder` instead of here (where I was following the analogous branch above this one), and there, those fields aren't set. I'll give it a go.